src/models/document/document-content-with-annotations.ts

```diff
@@ -0,0 +1,23 @@
+import { types } from "mobx-state-tree";
+
+import { DocumentContentModelWithTileDragging } from "./drag-tiles";
+import { ArrowAnnotation, ArrowAnnotationType } from "../annotations/arrow-annotation";
+
+/**
+ * This is one part of the DocumentContentModel. The other parts are
+ * DocumentContentModelWidthTileDragging and BaseDocumentContentModel.
+ * It was split out to reduce the size of the DocumentContentModel.
+ *
+ * This file should contain the properties, views, and actions that are
+ * related to adorning documents (i.e. sparrows).
+ */
+export const DocumentContentModelWithAnnotations = DocumentContentModelWithTileDragging
+  .named("DocumentContentModelWithAnnotations")
+  .props({
+    annotations: types.array(ArrowAnnotation)
```

Comment:
I think changing to map is a good call. I was following the pattern for objects in the drawing tile, but it seems like this is a low cost bit of insurance in case we do want to access annotations in O(1) time in the future. I also remember that MST arrays can be a bit finicky sometimes.

I know that order is now guaranteed for javascript objects, but is this also true for MST maps? I'm still not sure of the exact relationship between the two, or MST arrays vs plain JS arrays for that matter.